docs/deployment/custom_auth.md

```diff
@@ -0,0 +1,48 @@
+# Custom authentication
+
+It is possible to provide custom Zero Trust Authentication (ZTA) inside Livebook's Docker images.
+
+To do so, you must define a file with the `.exs` extension inside the `/app/extensions` of your Livebook image, for example, `/app/extensions/my_auth.exs`. This file should define at least one module, which implements the ZTA skeleton below:
+
+```elixir
+defmodule MyAuth do
+  use GenServer
+
+  @spec start_link(keyword) :: {:ok, pid()}
+  def start_link(opts) do
+    identity_key = opts[:identity_key]
+    GenServer.start_link(__MODULE__, identity_key, Keyword.take(opts, [:name]))
+  end
+
+  @spec authenticate(GenServer.server(), Plug.Conn.t(), keyword()) ::
+          {Plug.Conn.t(), map() | nil}
+  def authenticate(server, conn, opts \\ []) do
```

Comment:
The approach looks good, the only issue I see is dependencies. Someone may use `Req` (the most likely), then if we bump it in Livebook it's can break their code.

Reply:
I will add a note about this. Once we extract ZTA, we can say they can use both req and jose packages according to the nimble_zta package but for now I will explicitly list them.